parse sql host address and port

esara opened the pull request:
parse sql host address and port for client spans to address https://github.com/grafana/beyla/issues/1254

grcevski commented:
I really like the idea, I think we are almost there. I pulled this branch and I'm going to take a look at what it looks like to add this. I think the main issue is that `ongoing_server_connections` is only set for HTTP/gRPC server requests, but it will not have the outgoing DB information.

grcevski commented:
Hi @esara, I added a PR (on your branch) which should address the missing connection info and will work even if the SQL was done on another Go routine. https://github.com/esara/beyla/pull/14. If it works for you, after you accept the changes we can work on an integration test.

When I run this on the Go SQL test I added locally, it now shows something like:
```
2024-10-16 16:25:52.101642552 (527.375µs[527.375µs]) SQLClient 0 SELECT accounting.contacts [127.0.0.1 as 127.0.0.1:52456]->[127.0.0.1 as 127.0.0.1:5432] size:0B svc=[qosql go] traceparent=[00-bade493e98d04f9af022c4e64f3e31ba-b20d20f870017cfd[9a02c11b6f5334ca]-01]
2024-10-16 16:25:52.101642552 (1.17105ms[1.031505ms]) HTTP 200 GET /psqltest [::1 as ::1:41798]->[::1 as ::1:8080] size:0B svc=[qosql go] traceparent=[00-bade493e98d04f9af022c4e64f3e31ba-9a02c11b6f5334ca[0000000000000000]-01]
```

esara commented:
> Hi @esara, I added a PR (on your branch) which should address the missing connection info and will work even if the SQL was done on another Go routine. [esara#14](https://github.com/esara/beyla/pull/14). If it works for you, after you accept the changes we can work on an integration test.
> 
> When I run this on the Go SQL test I added locally, it now shows something like:
> 
> ```
> 2024-10-16 16:25:52.101642552 (527.375µs[527.375µs]) SQLClient 0 SELECT accounting.contacts [127.0.0.1 as 127.0.0.1:52456]->[127.0.0.1 as 127.0.0.1:5432] size:0B svc=[qosql go] traceparent=[00-bade493e98d04f9af022c4e64f3e31ba-b20d20f870017cfd[9a02c11b6f5334ca]-01]
> 2024-10-16 16:25:52.101642552 (1.17105ms[1.031505ms]) HTTP 200 GET /psqltest [::1 as ::1:41798]->[::1 as ::1:8080] size:0B svc=[qosql go] traceparent=[00-bade493e98d04f9af022c4e64f3e31ba-9a02c11b6f5334ca[0000000000000000]-01]
> ```

hi @grcevski this change looks great, I tested it locally with a slightly larger go application and a postgres backend and it is correctly detecting the sql service name and port and populates it in the client span

I am running more tests with the snowflake example, I mentioned in the issue later

grcevski commented:
Hi Endre, I made a new PR with some integration tests https://github.com/esara/beyla/pull/15. We've broken something so I made a draft PR so I can investigate the failrures myself. I'll push a new change to your PR once I know what's broken and close my PR.

grcevski commented:
Hi Endre, I pushed new changes to my branch which is also used for my new PR against your branch and the tests passed: https://github.com/grafana/beyla/pull/1265. If you accept my PR changes as-is I think we should be in a good shape. I wonder if all these failures here might have to do with the version of clang/llvm you use for producing the BPF binaries? We are working towards switching to docker based builds for the BPF binaries to avoid this in the future.

grcevski reviewed (approved):
LGTM! Thanks again @esara for your contribution!
